src/main/scala/com.snowplowanalytics.snowplow.micro/Run.scala

```diff
@@ -0,0 +1,176 @@
+import cats.effect.{ExitCode, IO, Resource, Sync}
+import cats.implicits._
+import com.monovore.decline.Opts
+import com.snowplowanalytics.iglu.client.resolver.registries.JavaNetRegistryLookup
+import com.snowplowanalytics.snowplow.badrows.Processor
+import com.snowplowanalytics.snowplow.collector.core._
+import com.snowplowanalytics.snowplow.collector.core.model.Sinks
+import com.snowplowanalytics.snowplow.enrich.common.enrichments.EnrichmentRegistry
+import com.snowplowanalytics.snowplow.enrich.common.enrichments.registry.{Enrichment, EnrichmentConf}
+import com.snowplowanalytics.snowplow.enrich.common.utils.{HttpClient, ShiftExecution}
+import com.snowplowanalytics.snowplow.micro.Configuration.MicroConfig
+import org.http4s.ember.client.EmberClientBuilder
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import java.io.File
+import java.security.{KeyStore, SecureRandom}
+import java.util.concurrent.Executors
+import javax.net.ssl.{KeyManagerFactory, SSLContext, TrustManagerFactory}
+import scala.concurrent.ExecutionContext
+import scala.sys.process._
+
+object Run {
+
+  implicit private def logger: Logger[IO] = Slf4jLogger.getLogger[IO]
+
+  def run(): Opts[IO[ExitCode]] = {
+    Configuration.load().map { configuration =>
+      handleAppErrors {
+        configuration
+          .flatTap(config => checkLicense(config.collector.license.accept))
+          .semiflatMap { validMicroConfig =>
+            buildEnvironment(validMicroConfig)
+              .use(_ => IO.never)
+              .as(ExitCode.Success)
+          }
+      }
+    }
+  }
+
+  private def buildEnvironment(config: MicroConfig): Resource[IO, Unit] = {
+    for {
+      _ <- Resource.eval(setupSSLContext())
+      enrichmentRegistry <- buildEnrichmentRegistry(config.enrichmentsConfig)
+      badProcessor = Processor(BuildInfo.name, BuildInfo.version)
+      adapterRegistry = MicroAdapterRegistry.create() 
+      lookup = JavaNetRegistryLookup.ioLookupInstance[IO]
+      sink = new MemorySink(config.iglu.client, lookup, enrichmentRegistry, config.outputEnrichedTsv, badProcessor, adapterRegistry)
+      collectorService = new Service[IO](
+        config.collector,
+        Sinks(sink, sink),
+        BuildInfo
+      )
+      collectorRoutes = new Routes[IO](
+        config.collector.enableDefaultRedirect,
+        config.collector.rootResponse.enabled,
+        config.collector.crossDomain.enabled,
+        collectorService
+      ).value
+
+      miniRoutes = new Routing(config.iglu.resolver)(lookup).value
+      allRoutes = miniRoutes <+> collectorRoutes
+      _ <- HttpServer.build[IO](
+        allRoutes,
+        if (config.collector.ssl.enable) config.collector.ssl.port else config.collector.port,
+        config.collector.ssl.enable,
+        config.collector.hsts,
+        config.collector.networking,
+        config.collector.monitoring.metrics
+      )
+    } yield ()
+  }
+
+  private def setupSSLContext(): IO[Unit] = IO {
```

Comment:
I've always found java keystores a bit pain to generate and manage. There is nothing wrong with this approach but I wonder if one day we should add `pem` support to ease on the user-side of SSL support. It shouldn't be big of a deal. Given you're AFK I can have a stab at it with the subsequent try work.